### DISCUSSION TITLE
The rate limit for refresh token requests was reached. Requests are blocked during the remaining period

### DISCUSSION THREAD

**Discussion by Dabbadoeber** [alexkn/node-red-contrib-homeconnect#50] [Q&A]
Is there a way we can reduced the number of api calls ?
we can have about 1000 (i have 3 devices)

is it possible to get the status of all devices with one call ?

**Comment by alexkn** [comment]
How do you generate 1000 API calls with 3 devices?
Please describe what you are currently doing. Maybe with a screenshot of your current flow.

**Reply by Hellis81** [reply]
If you update once every minute on one device on one endpoint you get 
1 x 60 x 24 = 1440 API calls per day.
Having three devices would be 
3 x 60 x 24 = 4320

Still just one endpoint.
But yes every minute is not required but still not impossible to reach

**Reply by alexkn** [reply]
Do not poll the status every minute.
Instead use one event node per device to get notified when something changes.

**Reply by Dabbadoeber** [reply]
it is mostly done, when i am deploying Node-red a lot of times, because of changes.
I think every time i deploy it will make a call.
